draft-ietf-taps-transport-security.md

```diff
@@ -854,170 +354,18 @@ configure and works over a variety of transports. OpenVPN encapsulates either
-
-- Application authentication delegation (AD):
-  - Application dependency: Application opt-in and policy for endpoint authentication.
-
-- DoS mitigation (DM):
-  - Application dependency: None.
-
-- Connection mobility (CM):
-  - Application dependency: None.
-
-- Source validation (SV):
-  - Application dependency: None.
-
-- Application-layer feature negotiation (AFN):
-  - Application dependency: Specification of application-layer features or functionality.
-
-- Configuration extensions (CX):
-  - Application dependency: Specification of application-specific extensions.
-
-- Session caching and management (SC):
-  - Application dependency: None.
-
-- Length-hiding padding (LHP):
-  (Optional) Application dependency: Knowledge of desired padding policies. Some protocols, such as IKE, can negotiate application-agnostic padding policies.
-
-- Early data support (ED):
-  - Application dependency: Anti-replay protections or hints of data idempotency.
-
-- Record replay prevention (RP):
-  - Application dependency: None.
-
-- Out-of-order receipt record (OO):
-  - Application dependency: None.
-
-## Optional Feature Availability
+# Transport Dependencies
 
-The following table lists the availability of the above-listed optional features in each of
-the analyzed protocols. "Mandatory" indicates that the feature is intrinsic to the protocol
-and cannot be disabled. "Supported" indicates that the feature is optionally provided natively
-or through a (standardized, where applicable) extension.
+- Stream
+Protocols: TLS, OpenVPN, tcpcrypt
 
-|---
-| Protocol  | PSK | AN | AD | MA | DM | CM | SV | AFN | CX | SC | LHP | ED | RP | OO |
-|:----------|:---:|:--:|:--:|:--:|:--:|:--:|:--:|:---:|:--:|:--:|:---:|:--:|:--:|:--:|
-| TLS       | S   | S  | S  | S  | S  | U\* | M | S   | S  | S  | S   | S  | U  | U  |
-| DTLS      | S   | S  | S  | S  | S  | S  | M  | S   | S  | S  | S   | U  | M  | M  |
-| QUIC | S   | S  | S  | S  | S  | S  | M  | S   | S  | S  | S   | S  | M  | M  |
-| IKEv2+ESP | S   | S  | S  | M  | S  | S  | M  | S   | S  | S  | S   | U  | M  | M  |
-| SRTP+DTLS | S   | S  | S  | S  | S  | U  | M  | S   | S  | S  | U   | U  | M  | M  |
-| tcpcrypt  | U   | S  | M  | U  | U\*\* | U\* | M | U | U | S  | U   | U  | U  | U  |
-| WireGuard | S   | U  | S  | M  | S  | U  | M  | U   | U  | U  | S+  | U  | M  | M  |
-| MinimalT  | U   | U  | U  | M  | S  | M  | M  | U   | U  | U  | S   | U  | U  | U  |
-| CurveCP   | U   | U  | U  | S  | S  | M  | M  | U   | U  | U  | S   | U  | M  | M  |
-|---
+- Datagram message framing
+Protocols: QUIC, DTLS, SRTP, IKEv2 and ESP, WireGuard, MinimalT, CurveCP
 
-M=Mandatory
-S=Supported but not required
-U=Unsupported
-\*=On TCP; MPTCP would provide this ability
-\*\*=TCP provides SYN cookies natively, but these are not cryptographically strong
-+=For transport packets only
+- Protocol-specific needs
+Protocols: tcpcrypt (TCP-ENO)
 
-# Transport Security Protocol Interfaces
+# Application Interface
```

Comment:
I think we also wanted some additional text here on how an application can explicitly choose a specific security protocol, how this needs to be explicit because the application should be aware of the limitations of the protocol, but how it should also be possible for an application to say that it considers multiple security protocol as equivalent.
So I'd put a TODO marker here.

Reply:
Sure, we can add a TODO here.

Reply:
I'm describing this in the goals/non-goals